google-cloud-clients/google-cloud-firestore/src/test/java/com/google/cloud/firestore/QueryTest.java

```diff
@@ -505,6 +497,43 @@ public void withEndAt() throws Exception {
     assertEquals(queryRequest, runQuery.getValue());
   }
 
+  @Test
+  public void withCollectionGroup() throws Exception {
+    doAnswer(queryResponse())
+        .when(firestoreMock)
+        .streamRequest(
+            runQuery.capture(),
+            streamObserverCapture.capture(),
+            Matchers.<ServerStreamingCallable>any());
+
+    Query query = firestoreMock.collectionGroup(COLLECTION_ID);
+    query = query.whereGreaterThan(FieldPath.documentId(), "coll/doc");
+    query = query.orderBy(FieldPath.documentId());
+    query = query.endAt("coll/doc");
+    query.get();
+
+    RunQueryRequest queryRequest =
+        query(
+            /* transactionId= */ null,
+            /* allDescendants= */ true,
+            filter(Operator.GREATER_THAN, "__name__", reference(DOCUMENT_NAME)),
+            order("__name__", StructuredQuery.Direction.ASCENDING),
+            endAt(reference(DOCUMENT_NAME), false));
+
+    assertEquals(queryRequest, runQuery.getValue());
+  }
+
+  @Test
+  public void collectionGroupCannotContainSlashes() {
+    try {
+      Query query = firestoreMock.collectionGroup("foo/bar");
+      fail();
+    } catch (IllegalArgumentException e) {
+      assertEquals(
```

Comment:
This is fine. Personally, I find it slightly over-specified. (I'd just remove the assertEquals entirely.) Feel free to leave as is.

Reply:
Acknowledged.